### DISCUSSION TITLE
How to build/output a model like FastAPI does with response_model

### DISCUSSION THREAD

**Discussion by blackhillsystems** [fastapi/sqlmodel#794] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from typing import Optional
from sqlmodel import Field, Session, SQLModel, create_engine, select

class Hero(SQLModel, table=True):
    id: Optional[int] = Field(default=None, primary_key=True)
    name: str = Field(index=True)
    secret_name: str
    age: Optional[int] = Field(default=None, index=True)
```

### Description

So, I build a generic model which imports from SQLModel. I've traced SQLModel back and it inherits Pydantic's BaseModel. Perfect. From there, I'm attempting to output (to OUTPUT, to UI, to anything - doesn't matter) the class back to the user much like is done in FastAPI's schema. For example, the FastAPI page would show:

Schemas
  Hero
    id* integer
    name str
    secret_name str
    age* integer

Or something like that. How is this achieved?  I thought I could just use jsonable_encoder on Hero, but it gives me the fields with what they're filled with (which makes sense) - e.g., None with an empty model.  So, then I thought I could rip through the model's fields and use SQLModel's get_sqlalchemy_type, but I can't access it from Hero since this method is outside of the SQLModel class (though in the SQLModel module). Any thoughts on the proper way to convert my Hero into JSON (or whatever) that provides the name of the field and it's type? I can just create my own, but I'm assuming there's a built-in that already does this.

Thank you! 

### Operating System

macOS

### SQLModel Version

0.103.1

### Python Version

3.11.5

**Comment by riziles** [comment, marked as answer]
@blackhillsystems , I'm not 100% sure I understand what you are asking. Does `Hero.schema()` or `Hero.schema_json()` give you what you are looking for?

**Comment by blackhillsystems** [comment]
How did I miss that!?!?!?  That'll work perfectly!  Thank you @riziles !
